src/extract-cli.js

```diff
@@ -67,7 +67,13 @@ files.forEach(function(filename) {
   try {
     let data = fs.readFileSync(file, {encoding: 'utf-8'}).toString();
     extractor.parse(file, extract.preprocessTemplate(data, ext));
-    extractor.parseJavascript(file, extract.preprocessJavascript(data, ext));
+    let script = extract.preprocessJavascript(data, ext)
+
+    if (script && script.lang === 'ts' ) {
+      extractor.parseTypeScript(file, script.content);
+    } else {
+      extractor.parseJavascript(file, data);
```

Comment:
it really seems to me that you should give `script` instead of `data` here. Or you'll break the whole javascript decoding feature.

Reply:
Thanks, this was unintentional